src/util/sss_iobuf.h

```diff
@@ -102,6 +112,11 @@ size_t sss_iobuf_get_size(const struct sss_iobuf *iobuf);
  */
 uint8_t *sss_iobuf_get_data(const struct sss_iobuf *iobuf);
 
+/*
+ * @brief Returns whether the data is secured
+ */
+bool sss_iobuf_get_secure(const struct sss_iobuf *iobuf);
```

Comment:
What would be the use?

Reply:
This is "what does it do", not "what would be a reason to use it".

What is the reason to store 'sss_iobuf::secure' at all?
When buffer is being created, one opts if he wants a secure cleanup.
But I can imagine a realistic use case when one wants to know if buffer is secured.

Reply:
The reason is explained in the last sentence:
> there is no other way for a caller to access this value.

The `struct sss_iobuf ` is defined in the `sss_iobuf.h` header file, but declared in the `sss_iobuf.c` module. External users only have access to those fields by using getters. As said before, today it is not used, but we don' t know if it will be need anytime soon. Also it was created for API completeness: all the other fields have a getter, this one too.

The `secure` field is needed by the `sss_iobuf_read_varlen()` and `sss_iobuf_read_iobuf()` functions which create copies of the data and need to know whether to secure or not those new buffers.

Reply:
> The `secure` field is needed by the `sss_iobuf_read_varlen()` and `sss_iobuf_read_iobuf()` functions which create copies of the data and need to know whether to secure or not those new buffers.

Could you please elaborate on this?

Reply:
Function `sss_iobuf_read_varlen()` receives an `sss_iobuf`, read some data from it and puts them on a brand new `sss_iobuf`. If the old structure were the data was read from was secured (`secure` is `true`), then the new one must also be secured. The cleanest way to know whether the original structure was secured is using the `secure` flag. This function is the reason the flag exists. Until I found this function, there was no flag.

Function `sss_iobuf_read_iobuf()` internally uses `sss_iobuf_read_varlen()` and `sss_iobuf_init_steal()` to create a new buffer, which must be secured in the same way, so the flag stored in the source structure is used to set the flag in the new structure. The flag isn't actually needed for this case, but since it exists for the other case, it must be set to the correct value.

Reply:
> This output buffer might be copied further, and information about it's "security" is already lost anyway. I would argue it's up to a user of `sss_iobuf_read_varlen()` to decide if read chunk of data has to be secured. This is more transparent.

No option is better than the other. I just choose the one that simplifies thing by letting the original owner of the data decide about its security. _"If it was originally secure, then it must remain secure."_

About further copies... Sure. There is nothing we can do about that right now.

> As to talloc_steal(), I believe it keeps d-tor.

Yes it does. I confirmed it. That's why I say `sss_iobuf_init_steal()` doesn't need the flag, but we need to keep it coherent for `sss_iobuf_read_varlen()`.

Reply:
Ok, I will not insist further, that's not too important after all (though I still think sss_iobuf_get_secure()/secure isn't really needed).